examples/java/native-assets-example/src/main/java/iog/psg/assets/Scenarios.java

```diff
@@ -0,0 +1,111 @@
+package iog.psg.assets;
+
+import com.bloxbean.cardano.client.address.Address;
+import com.bloxbean.cardano.client.common.model.Networks;
+import com.bloxbean.cardano.client.crypto.Keys;
+import com.bloxbean.cardano.client.exception.CborSerializationException;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.grpc.stub.StreamObserver;
+import iog.psg.client.nativeassets.NativeAssetsApi;
+import iog.psg.client.nativeassets.NativeAssetsBuilder;
+import iog.psg.client.nativeassets.multisig.v1.NativeAssetsMultisigApi;
+import iog.psg.service.nativeassets.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+public class Scenarios {
+
+    public static <E, R> StreamObserver<E> lastResult(CompletableFuture<R> result, Function<E, R> f) {
+        return new StreamObserver<>() {
+            ArrayList<R> buff = new ArrayList();
+
+            @Override
+            public void onNext(E event) {
+                R converted = f.apply(event);
+                System.out.println("Got interim response: " + converted);
+                buff.add(converted);
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                System.out.println("Got error: " + throwable.getCause());
+            }
+
+            @Override
+            public void onCompleted() {
+                result.complete(buff.get(buff.size() - 1));
+            }
+        };
+
+    }
+
+    public static void scenario1()  throws ExecutionException, InterruptedException, CborSerializationException  {
```

Comment:
At least we need to define scenario as a method name.

Reply:
renamed